File: app/src/main/java/com/funckyhacker/fileexplorer/di/AppComponent.java
package com.funckyhacker.fileexplorer.di;

import com.funckyhacker.fileexplorer.App;
import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
    AndroidSupportInjectionModule.class,
    AppModule.class,
    MainModule.class,
    SearchModule.class

})
public interface AppComponent extends AndroidInjector<App> {

  @Component.Builder
  abstract class Builder extends AndroidInjector.Builder<App> {
  }
}

File: app/src/main/java/com/funckyhacker/fileexplorer/di/AppComponent.kt
package com.funckyhacker.fileexplorer.di

import com.funckyhacker.fileexplorer.App
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(modules = [
    (AndroidSupportInjectionModule::class),
    (AppModule::class),
    (MainModule::class),
    (SearchModule::class)
])
interface AppComponent : AndroidInjector<App> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<App>()
}
